kuma/javascript/src/task-completion-survey.jsx

```diff
@@ -0,0 +1,193 @@
+            // send these GA events each time we re-render.
+            ga('set', 'dimension14', 'Yes');
+            ga('send', {
+                hitType: 'event',
+                eventCategory: 'survey',
+                eventAction: 'prompt',
+                eventLabel: '',
+                eventValue: 'impression',
+                nonInteraction: true
+            });
+        }
+    }, [shouldShow]);
+
+    // If we're not going to show anything, we can just return now.
+    if (!shouldShow) {
+        return null;
+    }
+
+    // Construct the survey URL.
+    let surveyURL = 'https://www.surveygizmo.com/s3/2980494/do-or-do-not';
+    // Use a future timestamp because we're asking the user to open
+    // the survey now and fill it out later.  20 minutes is a guess.
+    surveyURL += `?t=${Date.now() + 1000 * 60 * 20}`;
+    if (documentData) {
+        surveyURL += `&p=${encodeURIComponent(
+            `/${locale}/docs/${documentData.slug}`
+        )}`;
+    }
+    // When we are first displayed we won't have the GA client ID, but the
+    // useClientId() hook function above will get it and we'll be re-rendered
+    // which will update the URL to add the client id.
+    if (clientId) {
+        surveyURL += `&c=${encodeURIComponent(clientId)}`;
+    }
+
+    // Called when the user clicks the X icon or the survey link
+    function dismiss() {
+        // When the notification is dismissed ensure that the user
+        // won't see it again for a period of time.
+        embargo();
+        // And now set the state variable that actually hides the notification.
+        setDismissed(true);
+    }
+
+    // Called when the user clicks the survey link
+    function recordAndDismiss() {
+        // If the user clicks on the survey link, let Google Analytics know.
+        ga('send', {
+            hitType: 'event',
+            eventCategory: 'survey',
+            eventAction: 'prompt',
+            eventLabel: '',
+            eventValue: 'participate',
+            nonInteraction: false
+        });
+        // Since they opened the survey, we can dismiss the notification
+        dismiss();
+    }
+
+    return (
+        <div css={styles.notification}>
+            <div>
+                {gettext('Would you answer 4 questions for us?')}{' '}
+                <a
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    href={surveyURL}
+                    onClick={recordAndDismiss}
+                >
+                    {gettext('Open the survey in a new tab')}
+                </a>{' '}
+                {gettext(
+                    'and fill it out when you are done on the site. Thanks!'
+                )}
+            </div>
+            <button
+                css={styles.dismissButton}
+                onClick={dismiss}
+                aria-label={gettext('Dismiss')}
+            >
+                <CloseIcon css={styles.dismissIcon} role="presentation" />
```

Comment:
@davidflanagan - Is there a reason to not add the `role` directly to the SVG? Seems unlikely that we will ever use the SVG in a scenario where the `role` will not be valid.

Reply:
I tried that first, but the role attribute was getting stripped out, presumably by svg-react-loader. I didn't investigate why it was happening, and just move the role attribute here instead.

Reply:
react-svg-loader invokes svgo, and it appears that svgo is stripping out the role attribute on the svg element.

Reply:
Despite that fact that react-svg-loader and svgo have terrible documentation, I was able to figure out how to pass svgo config settings to react-svg-loader in the webpack config file, and I disabled the svgo plugin that was stripping the role attribute, so I'm able to remove it here.